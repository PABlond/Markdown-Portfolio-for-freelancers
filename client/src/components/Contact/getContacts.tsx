import React, { useState } from "react"
import store from "./../../state/store"
import { Container, Row, Accordion, Card, Col, Button } from "react-bootstrap"
import moment from "moment"
import readMessage from "./../../services/readMessage"
import dispatchFullContent from "./../../state/actions/dispatchFullContent"
import { IContact } from "../../interfaces/contact.interface"

export default () => {
  const props = store.getState()
  const { works, about, header, contacts } = props.content
  const [contactsState, setContactsState] = useState<IContact[]>(contacts)

  const onClick: (i: number) => any = async i => {
    if (!contactsState[i].isRead) {
      const { _id } = contactsState[i]
      const newContacts: { contact: IContact[] } = await readMessage({ _id })
      dispatchFullContent({ contact: newContacts, works, about, header })
      setContactsState(newContacts.contact)
    }
  }

  return (
    <Container className="mt-3 p-5">
      <h2 className="text-center mb-4 text-danger">Last form submissions</h2>
      <Accordion defaultActiveKey="0">
        {contactsState.reverse().map((contact: IContact, i: number) => (
          <Card key={i} onClick={() => onClick(i)}>
            <Card.Header
              className={(!contact.isRead ? "bg-danger" : null) + " p-0"}
            >
              <Accordion.Toggle
                className={"w-100"}
                as={Button}
                variant="link"
                eventKey={i.toString()}
              >
                <Container>
                  <Row className={contact.isRead ? "text-dark" : "text-light"}>
                    <Col md={8}>
                      <p className="m-0">{contact.name}</p>
                    </Col>
                    <Col md={4}>
                      <p className="m-0">
                        {moment
                          .unix(contact.at / 1000)
                          .format("MM/DD/YYYY h:mm:ss A")}
                      </p>
                    </Col>
                  </Row>
                </Container>
              </Accordion.Toggle>
            </Card.Header>
            <Accordion.Collapse eventKey={i.toString()}>
              <Card.Body>
                <p>{contact.content}</p>
              </Card.Body>
            </Accordion.Collapse>
          </Card>
        ))}
      </Accordion>
    </Container>
  )
}
