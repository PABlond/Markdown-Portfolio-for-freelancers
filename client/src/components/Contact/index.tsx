import React, { useState } from "react"
import store from "./../../state/store"
import { Container, Form, Button } from "react-bootstrap"
import sendMessage from "./../../services/sendMessage"

export default () => {
  const [form, setValues] = useState({
    email: "",
    name: "",
    content: "",
  })
  const [response, setResponse] = useState("")

  const onSubmit = async (e: any) => {
    e.preventDefault()
    setResponse(await sendMessage(form))
    setValues({
      email: "",
      name: "",
      content: "",
    })
  }

  const updateField = (e: any) =>
    setValues({
      ...form,
      [e.target.name]: e.target.value,
    })

  return (
    <Container id="contact" fluid>
      <h1>Want to hire me ?</h1>
      <Form onSubmit={onSubmit}>
        <Form.Group>
          <Form.Label>Email</Form.Label>
          <Form.Control
            type="email"
            name="email"
            placeholder="name@example.com"
            value={form.email}
            onChange={updateField}
          />
        </Form.Group>
        <Form.Group>
          <Form.Label>Name</Form.Label>
          <Form.Control
            type="text"
            name="name"
            placeholder="John Doe"
            value={form.name}
            onChange={updateField}
          />
        </Form.Group>
        <Form.Group>
          <Form.Label>Message</Form.Label>
          <Form.Control
            as="textarea"
            name="content"
            rows="3"
            value={form.content}
            onChange={updateField}
          />
        </Form.Group>
        <Button
        block
          type="submit"
          id="contact-button"
          variant="danger"
        >
          SUBMIT
        </Button>
        {response.length ? <p className="text-success">{response}</p> : null}
      </Form>
    </Container>
  )
}
