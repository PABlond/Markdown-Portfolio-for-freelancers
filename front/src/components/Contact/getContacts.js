import React, { useState } from "react"
import store from "./../../state/store"
import { Container, Row, Accordion, Card, Col, Button } from "react-bootstrap"
import moment from "moment"
import readMessage from './../../services/readMessage'
import dispatchFullContent from './../../state/actions/dispatchFullContent'

export default () => {
  const props = store.getState()
  const { works, about, header, contacts } = props.content
  console.log(contacts)
  const [contactsState, setContactsState] = useState(contacts)

  const onClick = async i => {
      if (!contactsState[i].isRead) {
        const {_id} = contactsState[i]
        const newContacts = await readMessage({_id})
        dispatchFullContent({contact: newContacts, works, about, header})
        setContactsState(newContacts.contact)
      }
  }

  return (
    <Container>
      <Accordion defaultActiveKey="0">
        {contactsState.map((contact, i) => (
          <Card key={i} onClick={() => onClick(i)}>
            <Card.Header className={!contact.isRead ? "bg-danger" : null}>
              <Accordion.Toggle as={Button} variant="link" eventKey={i}>
                <Row>
                <Col md={4}>
                  <p>{contact.email}</p>
                </Col>
                <Col md={4}>
                  <p>{contact.name}</p>
                </Col>
                <Col md={4}>
                  <p>
                    {moment
                      .unix(contact.at / 1000)
                      .format("MM/DD/YYYY h:mm:ss A")}
                  </p>
                </Col>
                </Row>
              </Accordion.Toggle>
            </Card.Header>
            <Accordion.Collapse eventKey={i}>
              <Card.Body>
                <p>{contact.content}</p>
              </Card.Body>
            </Accordion.Collapse>
          </Card>
        ))}
      </Accordion>
    </Container>
  )
}
