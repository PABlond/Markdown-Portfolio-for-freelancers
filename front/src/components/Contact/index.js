import React, { useState } from "react"
import store from "./../../state/store"
import { Container, Form, Button } from "react-bootstrap"
import { contact as contactStyle } from "./../../styles/style"
import sendMessage from './../../services/sendMessage'

export default ({ content }) => {
  const [form, setValues] = useState({
    email: "",
    name: "",
    content: "",
  })
  
  const onSubmit = async e => {
    e.preventDefault()
    const response = await sendMessage(form)
  }

  const updateField = e =>
    setValues({
      ...form,
      [e.target.name]: e.target.value,
    })

  const { width, height } = store.getState().nav
  const style = contactStyle({ width, height })

  return (
    <Container fluid style={style.container}>
      <h1 style={style.h1}>Want to hire me ?</h1>
      <Form style={style.form} onSubmit={onSubmit}>
        <Form.Group style={style.group}>
          <Form.Label>Email</Form.Label>
          <Form.Control
            type="email"
            name="email"
            placeholder="name@example.com"
            value={form.email}
            onChange={updateField}
          />
        </Form.Group>
        <Form.Group style={style.group}>
          <Form.Label>Name</Form.Label>
          <Form.Control
            type="text"
            name="name"
            placeholder="John Doe"
            value={form.name}
            onChange={updateField}
          />
        </Form.Group>
        <Form.Group style={style.group}>
          <Form.Label>Message</Form.Label>
          <Form.Control
            as="textarea"
            name="content"
            rows="3"
            value={form.content}
            onChange={updateField}
          />
        </Form.Group>
         <Button type="submit" id="contact-button" variant="danger" style={style.submit}>
            SUBMIT
          </Button>
      </Form>
    </Container>
  )
}
