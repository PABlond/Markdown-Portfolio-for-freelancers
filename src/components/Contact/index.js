import React, { useState } from "react"
import axios from "axios"
import pageStructure from '../App/pageStructure'

export default ({ content }) => {
  const [form, setValues] = useState({
    email: "",
    name: "",
    content: "",
  })
  const [APIRes, setAPIRes] = useState({
    isError: false,
    content: "",
  })

  const handleClick = async e => {
    e.preventDefault()
    try {
      const response = await axios.post(
        "http://localhost:1337/api/contact",
        form
      )

      setAPIRes({
        isError: false,
        content: "An error occured. Feel free to contact me directly",
      })
    } catch (error) {
      console.error(error)
      setAPIRes({
        isError: true,
        content: "An error occured. Feel free to contact me directly",
      })
    }
  }

  const updateField = e =>
    setValues({
      ...form,
      [e.target.name]: e.target.value,
    })

  return (
    <div id={pageStructure[content.n].id}>
      <div id="content-heading" dangerouslySetInnerHTML={content} />
      <div id="contact-form">
        <form onSubmit={handleClick}>
          <input
            type="text"
            name="email"
            placeholder="Email"
            value={form.email}
            onChange={updateField}
          />
          <input
            type="text"
            name="name"
            placeholder="Name"
            value={form.name}
            onChange={updateField}
          />
          <textarea
            name="content"
            placeholder="Your message"
            rows={5}
            value={form.content}
            onChange={updateField}
          />

          <input type="submit" />
          <p className={APIRes.isError ? "red" : "green"}>{APIRes.content}</p>
        </form>
      </div>
    </div>
  )
}
