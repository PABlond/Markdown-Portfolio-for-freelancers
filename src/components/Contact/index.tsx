import React, { useState } from "react"
import { Container, Form, Button } from "react-bootstrap"

export default () => {
  const [form, setValues] = useState({
    email: "",
    name: "",
    content: "",
  })

  const encode = (data: any) => {
    return Object.keys(data)
      .map(key => encodeURIComponent(key) + "=" + encodeURIComponent(data[key]))
      .join("&")
  }

  const onSubmit = (e: any) => {
    e.preventDefault()

    fetch("/", {
      method: "POST",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: encode({ "form-name": "contact", ...form }),
    })
      .then(() => alert("Success!"))
      .catch(error => alert(error))

    setValues({
      email: "",
      name: "",
      content: "",
    })
  }

  const updateField = (e: any) =>
    setValues({
      ...form,
      [e.target.name]: e.target.value,
    })

  return (
    <Container fluid>
      <h1>Want to hire me ?</h1>
      <Form onSubmit={onSubmit} className="offset-md-3 col-md-6">
        <Form.Group>
          <Form.Label>Email</Form.Label>
          <Form.Control
            type="email"
            name="email"
            placeholder="name@example.com"
            value={form.email}
            onChange={updateField}
          />
        </Form.Group>
        <Form.Group>
          <Form.Label>Name</Form.Label>
          <Form.Control
            type="text"
            name="name"
            placeholder="John Doe"
            value={form.name}
            onChange={updateField}
          />
        </Form.Group>
        <Form.Group>
          <Form.Label>Message</Form.Label>
          <Form.Control
            as="textarea"
            name="content"
            rows="3"
            value={form.content}
            onChange={updateField}
          />
        </Form.Group>
        <Button block type="submit" id="contact-button" variant="danger">
          SUBMIT
        </Button>
      </Form>
    </Container>
  )
}
