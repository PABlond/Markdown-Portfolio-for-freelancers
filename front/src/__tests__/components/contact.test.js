import React from "react"
import renderer from "react-test-renderer"
import { shallow, configure } from "enzyme"
import Contact from "../../components/Contact"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

const constactShallowCfg = () => {
  const props = {
    content: { n: "1" },
  }
  const enzymeWrapper = shallow(<Contact {...props} />)
  return {
    props,
    enzymeWrapper,
  }
}

describe("Shallow rendered Contact", () => {
  it("should render match snapshot", () => {
    const component = renderer.create(<Contact />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("should render contact as expected", () => {
    const { enzymeWrapper } = constactShallowCfg()
    expect(enzymeWrapper.find("h1").text()).toBe("Want to hire me ?")
  })
})
