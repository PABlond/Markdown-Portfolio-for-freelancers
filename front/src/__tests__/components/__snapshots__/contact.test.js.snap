// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered Contact should render match snapshot 1`] = `
<div
  className="container-fluid"
  style={
    Object {
      "alignItems": "center",
      "background": "#61234e",
      "display": "flex",
      "flexDirection": "column",
      "height": "inherit",
      "justifyContent": "center",
    }
  }
>
  <h1
    style={
      Object {
        "color": "#ebfffb",
        "fontSize": "3rem",
        "marginBottom": "0",
        "marginTop": "20px",
        "textAlign": "center",
      }
    }
  >
    Want to hire me ?
  </h1>
  <form
    className=""
    onSubmit={[Function]}
    style={
      Object {
        "background": "#ebfffb",
        "borderRadius": "5px",
        "margin": "50px",
        "padding": "30px",
      }
    }
  >
    <div
      className="form-group"
      style={
        Object {
          "width": "50vw",
        }
      }
    >
      <label
        className="form-label"
      >
        Email
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        placeholder="name@example.com"
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
      style={
        Object {
          "width": "50vw",
        }
      }
    >
      <label
        className="form-label"
      >
        Name
      </label>
      <input
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="John Doe"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
      style={
        Object {
          "width": "50vw",
        }
      }
    >
      <label
        className="form-label"
      >
        Message
      </label>
      <textarea
        className="form-control"
        name="content"
        onChange={[Function]}
        rows="3"
        value=""
      />
    </div>
    <button
      className="btn btn-danger"
      disabled={false}
      id="contact-button"
      style={
        Object {
          "background": "#61234e!important",
          "width": "50vw",
        }
      }
      type="submit"
    >
      SUBMIT
    </button>
  </form>
</div>
`;
